onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -40,10 +41,72 @@ std::string GetEnginePath(const ::std::string& root, const std::string& name) {
   }
 }
 
-std::string GetVecHash(const ::std::vector<int>& vec) {
+std::string GetProfilePath(const ::std::string& root, const std::string& name) {
+  if (root.empty()) {
+    return name + ".profile";
+  } else {
+    fs::path path = root;
+    path.append(name + ".profile");
+    return path.string();
+  }
+}
+
+void SerializeProfile(const ::std::string& file_name, std::unordered_map<std::string, std::unordered_map<int, std::pair<int64_t, int64_t>>>& shape_ranges) {
```

Comment:
Can you please leave some detailed comments about how the Profile data is organized/stored?
and perhaps comment why we are going schema-less by going with FlexBuffers instead of FlatBuffers ?

Reply:
will we add some debug tooling to read the profiles?

Reply:
Sure. Will also explain it in TRT doc.

Reply:
Both flatbuffers and flexbuffers can do the job. The reason to use flexbuffers is the profile size is small and data structure is pretty simple, so schema is not really necessary